v07 Petar Puharic

PetarPuharic opened the pull request.

ncaklovic commented:
Sve u redu.

PetarPuharic committed:
Prvi commit dodani basic testovi i class implementacija

PetarPuharic committed:
.

ncaklovic commented:
Build nije prošao. Zašto se nije izvrtio [na vašem repou](https://github.com/PetarPuharic/v06/actions)?

PetarPuharic committed:
Popravak
Oprostite sto mi je trebalo dugo, problem je bio glup. Trebao sam u oba promjeniti c++ na najnoviju vrijednost, i neznam zasto ali unittest neradi bez .cpp

ncaklovic commented:
Ne, i dalje ne prolazi, zar ne vidite [na svom repou](https://github.com/PetarPuharic/v06/actions)?

PetarPuharic committed:
cpp17 na test

PetarPuharic committed:
Promejna Relese brancha

PetarPuharic committed:
FileListAbsolute

ncaklovic commented:
Promijenite u [skripti za build](https://github.com/PetarPuharic/v06/blob/main/.github/workflows/msbuild.yml#L38) naziv dll-a sa unit testovima - vaš projekt je UnitTest1 pa je i dll: `UnitTest1.dll`.

PetarPuharic committed:
Change test DLL in VSTest command

ncaklovic commented on app/money.cpp:
```diff
@@ -1,7 +1,61 @@
+			return os;
+		}
+		else {
+			if (cents < 10)
```
ne, koristite `std::format`

ncaklovic commented on UnitTest1/UnitTest1.cpp:
```diff
@@ -0,0 +1,61 @@
+
+// Neznam u cemu je problem, normal run radi bez ovoga
+// Ali build neradi bez ovoga
+#include "../app/money.cpp"
```
ne, .cpp fileovi se nikad ne _include_-aju - dodajte ga u projekt

ncaklovic commented on UnitTest1/UnitTest1.cpp:
```diff
@@ -0,0 +1,61 @@
+			std::ostringstream oss;
+			money m1(3, 5);
+			oss << m1;
+			Assert::AreEqual(std::string("3 eur, 05 ct"), oss.str());
```
možete koristiti literale (pogledajte prethodne vježbe) - `"3 eur, 05 ct"s`

ncaklovic commented on UnitTest1/UnitTest1.cpp:
```diff
@@ -0,0 +1,61 @@
+			money m1(3, 5);
+			oss << m1;
+			Assert::AreEqual(std::string("3 eur, 05 ct"), oss.str());
+		}
```
ovo je jedini test za ispis (a 0 eura, 0 centa?)

ncaklovic commented on v06 Test/v06 Test.cpp:
```diff
@@ -0,0 +1,16 @@
+	{
+	public:
+		
+		TEST_METHOD(TestMethod1)
```
šta je ovo? pobrišite suvišne fileove

PetarPuharic committed:
Dodani extra tests

PetarPuharic committed:
Dodani std::format

PetarPuharic committed:
maknio bezkorisni #include

PetarPuharic committed:
Merge branch 'main' of https://github.com/PetarPuharic/v06

ncaklovic commented:
Nekoliko stvari:
- unit test projekt u _release_ modu ne koristi _precompiled header_, a u _debug_ modu da (i onda imate suvišni `pch.cpp` u projektu)
- ispis kad je broj centi >= 10 uopće ne radi (dodajte test sa i bez eura)
- nije ideja formatiranja ispisa da radite `"0{}"` (očito krivo) nego da koristite _format specifier_ unutar zagrada (prva vježba)

PetarPuharic committed:
Maknio pch.cpp, uredio format

ncaklovic commented:
Ispis uopće ne radi (kad su centi veći od 10, a euri veći od 0). Dodajte test za to pa probajte (imate i _warning_ od compilera za to).

PetarPuharic committed:
Popravio if logiku kado da se sada pokazuju vrijednosti preko 10 centi

ncaklovic commented:
Sve u redu.
